connectors/es/management_client.py

```diff
@@ -99,6 +99,59 @@ async def ensure_content_index_mappings(self, index, mappings):
                 "Index %s already has mappings, skipping mappings creation", index
             )
 
+    async def ensure_content_index_settings(
+        self, index_name, index, language_code=None
+    ):
+        existing_settings = index.get("settings", {})
+        settings = Settings(language_code=language_code, analysis_icu=False).to_hash()
+
+        if "analysis" not in existing_settings.get("index", {}):
+            logger.debug(
+                f"Index {index_name} has no settings or it's empty. Adding settings..."
+            )
+
+            # Open index, update settings, close index
+            try:
+                if self.serverless:
+                    await self._retrier.execute_with_retry(
+                        partial(
+                            self.client.perform_request,
+                            "PUT",
+                            f"/{index_name}/_settings?reopen=true",
+                            body=settings,
+                            headers={
+                                "accept": "application/json",
+                                "content-type": "application/json",
+                            },
+                        )
+                    )
+                else:
+                    await self._retrier.execute_with_retry(
+                        partial(self.client.indices.close, index=index_name)
+                    )
+
+                    await self._retrier.execute_with_retry(
+                        partial(
+                            self.client.indices.put_settings,
+                            index=index_name,
+                            body=settings,
+                        )
+                    )
+
+                    await self._retrier.execute_with_retry(
+                        partial(self.client.indices.open, index=index_name)
+                    )
+            except Exception as e:
```

Comment:
Should we not terminate in this case?
The way this is written, your logs would look like:

```
<timestamp1> - WARN - Could not create settings for index {index_name}, encountered error {e}
<timestamp2> - DEBUG - Successfully added settings for index {index_name}
```

which seems wrong.